### ISSUE TITLE
Create a documentation distribution together with other release artifacts

### ISSUE THREAD

**Issue by ghost** [stratosphere/stratosphere#871]
It would be good to have a documentation distribution together with the other release artifacts. We can use markdown files, .md, for documentation and use maven md plugin for managements. The same documentation can be used for the web site etc..

**Comment by rmetzger** [comment]
I agree that we need to ship the documentation together with the distribution. This ensures that users always have a documentation that matches with the version we ship.

What we first need to agree upon is a central instance that contains the documentation. I suggest to use a Wiki for that. I would then use the build tools to synchronize the documentation from the Wiki to the website and the release (and snapshot) tars.
I think that we let the wiki evolve over time (so no separate pages for the releases) and create dumps for each release that make accessible separately.

The whole thing becomes even more complicated with our move to Apache.
Does anybody know if other projects have a similar setup?

**Comment by ghost** [comment]
At Apache, the documentation for most projects is maintained using couple of XML or MD files together with the source code it self, no wiki. We could follow the same model. We must make sure, we maintain the documentation only at a single place.

**Comment by rmetzger** [comment]
I think that is also a pretty good approach.
This way developers can include the updated documentation into the
patches/pull requests.

The reason why I suggested a Wiki is the even lower barrier to change the
documentation. With a Wiki, people that don't know git are also able to
help. I dont know if those people exist and if its worth the effort.

Sent from my iPhone

On 03.06.2014, at 03:23, Rajika Kumarasiri notifications@github.com wrote:

At Apache, the documentation for most projects is maintained using couple
of XML or MD files together with the source code it self, no wiki. We could
follow the same model. We must make sure, we maintain the documentation
only at a single place.
